[Inductor] Add CPU_MAX_FIRST_DIMENSION_DECOMPOSITION and CPU_MAX_OTHER_DIMENSION_DECOMPOSITION for decompose_mm_pass

hl475 opened the pull request:
Differential Revision: D78209993




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

houseroad commented on torch/_inductor/fx_passes/decompose_mem_bound_mm.py:
```diff
@@ -17,16 +17,30 @@
 
 min_first_dimension_decomposition = MIN_FIRST_DIMENSION_DECOMPOSITION
 max_other_dimension_decomposition = MAX_OTHER_DIMENSION_DECOMPOSITION
+cpu_max_first_dimension_decomposition = CPU_MAX_FIRST_DIMENSION_DECOMPOSITION
```
curious if we can reuse MIN_FIRST_DIMENSION_DECOMPOSITION?

hl475 replied:
I would prefer to create a new one for the CPU case as the existing MIN_FIRST_DIMENSION_DECOMPOSITION is mostly used for CUDA case, and the number could be changed/diverged in the future.

houseroad commented on torch/_inductor/fx_passes/decompose_mem_bound_mm.py:
```diff
@@ -17,16 +17,30 @@
 # TODO: need a better strategy for decomposing mm
 MIN_FIRST_DIMENSION_DECOMPOSITION = 10240
```
add some comment to indicate this is for GPU?

hl475 committed:
[Inductor] Add CPU_MAX_FIRST_DIMENSION_DECOMPOSITION and CPU_MAX_OTHER_DIMENSION_DECOMPOSITION for decompose_mm_pass (#158183)

Summary:

This diff will refactor decompose_mem_bound_mm.py a bit by adding two constants CPU_MAX_FIRST_DIMENSION_DECOMPOSITION and CPU_MAX_OTHER_DIMENSION_DECOMPOSITION by following what GPU does (MIN_FIRST_DIMENSION_DECOMPOSITION and MAX_OTHER_DIMENSION_DECOMPOSITION)

In this way, we can dynamically change the config from internal and validate so no need to modify this file from OSS any more. 

In addition, this diff changes the current number from 128 and 512 to 2048, where we validated that decompose will help the perf.

Test Plan:
Sandcastle

Rollback Plan:

Reviewed By: houseroad

Differential Revision: D78209993
